python/tests/nonreg/arcticdb/version_store/test_nonreg_specific.py

```diff
@@ -197,3 +198,155 @@ def test_batch_write_unicode_strings(lmdb_version_store):
+
+
+def test_update_with_empty_series_or_dataframe(lmdb_version_store):
+    # Non-regression test for https://github.com/man-group/ArcticDB/issues/892
+    lib = lmdb_version_store
+
+    empty_df = pd.DataFrame([], columns=["a", "b", "c"], dtype=float)
+    one_row_df = pd.DataFrame(
+        data=np.array([[1.0, 2.0, 3.0]]),
+        columns=["a", "b", "c"],
+        dtype=float,
+        index=pd.DatetimeIndex([
+            datetime.datetime(2019, 4, 9, 10, 5, 2, 1)
+        ]),
+    )
+
+    symbol = "test_update_with_empty_dataframe_first"
+
+    lib.write(symbol, empty_df)
+    lib.append(symbol, empty_df)
+    lib.update(symbol, one_row_df)
+    received_df = lib.read(symbol).data
+    assert_frame_equal(one_row_df, received_df)
+
+    symbol = "test_update_with_empty_dataframe_second"
+
+    lib.write(symbol, one_row_df)
+    if IS_PANDAS_TWO:
+        # Appending and updating with an empty dataframe must not have any effect but must not fail.
+        # TODO: Currently Padnas 1 default empty dtype series being corced to EMPTY is causing problems
+        # with changing the stored descriptors, making updating or reading symbol back impossible
+        # when empty dataframes are appended.
+        lib.append(symbol, empty_df)
+        lib.update(symbol, empty_df)
+
+    received_df = lib.read(symbol).data
+    assert_frame_equal(one_row_df, received_df)
+
+    empty_series = pd.Series([], dtype=float, index=pd.DatetimeIndex([]))
+    one_row_series = pd.Series([1.0], dtype=float, index=pd.DatetimeIndex([datetime.datetime(2019, 4, 9, 10, 5, 2, 1)]))
+
+    symbol = "test_update_with_empty_series_first"
+
+    lib.write(symbol, empty_series)
+    lib.append(symbol, empty_series)
+    lib.update(symbol, one_row_series)
+    received_series = lib.read(symbol).data
+    assert_series_equal(one_row_series, received_series)
+
+    symbol = "test_update_with_empty_series_second"
+
+    lib.write(symbol, one_row_series)
+    if IS_PANDAS_TWO:
+        # Appending and updating with an empty series must not have any effect but must not fail.
+        # TODO: Currently Padnas 1 default empty dtype series being corced to EMPTY is causing problems
+        # with changing the stored descriptors, making updating or reading symbol back impossible
+        # when empty series are appended.
+        lib.append(symbol, empty_series)
+        lib.update(symbol, empty_series)
+
+    received_series = lib.read(symbol).data
+    assert_series_equal(one_row_series, received_series)
+
+
+def test_update_with_empty_dataframe_with_index(lmdb_version_store):
+    # Non-regression test for https://github.com/man-group/ArcticDB/issues/940
+    lib = lmdb_version_store
+
+    symbol = "test_update_with_empty_dataframe_with_index"
+
+    series = pd.Series(dtype="datetime64[ns]")
+    lib.write(symbol, series)
+
+    # This must not fail.
+    lib.read(symbol, as_of=0).data
+
+
+def test_empty_column_handling(lmdb_version_store):
```

Comment:
This tests seems rather long to me, can we split it or use some fixture to make it a bit more readable?

Reply:
Done with ff99224.